provider/aws/iam.go

```diff
@@ -0,0 +1,164 @@
+package aws
+
+import (
+	"finala/expression"
+	"finala/storage"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/iam"
+	log "github.com/sirupsen/logrus"
+)
+
+// IAMClientDescreptor is an interface of IAM client
+type IAMClientDescreptor interface {
+	ListUsers(input *iam.ListUsersInput) (*iam.ListUsersOutput, error)
+	ListAccessKeys(input *iam.ListAccessKeysInput) (*iam.ListAccessKeysOutput, error)
+	GetAccessKeyLastUsed(input *iam.GetAccessKeyLastUsedInput) (*iam.GetAccessKeyLastUsedOutput, error)
+}
+
+// IAMManager describe the iam manager
+type IAMManager struct {
+	client           IAMClientDescreptor
+	storage          storage.Storage
+	cloudWatchClient *CloudwatchManager
```

Comment:
I don't see you are using these variables , why do you need them ?
cloudWatchClient
pricingclient
region